### ISSUE TITLE
sync permissions and prompt to overwrite

### ISSUE THREAD

**Issue by jobevers** [Azure/azure-storage-azcopy#208]
### Which version of the AzCopy was used? 

AzCopy/10.0.7-Preview Azure-Storage/0.4

### Which platform are you using? (ex: Windows, Mac, Linux)

Linux

### What command did you run?

```
azcopy sync . https://<account>.blob.core.windows.net/<container>/<path>/ --recursive --log-level INFO
```

### What problem was encountered?

1) Permissions are weird. I ran it three times.  The first time I got this error:

```
Using OAuth token for authentication.

Job 6974d566-47b2-e642-5b96-3ac2527133ea has started

Log file is located at: /home/jobevers/.azcopy/6974d566-47b2-e642-5b96-3ac2527133ea.log
0 File Scanned at Source, 0 Files Scanned at Destination
Authentication Failed. The SAS is not correct or expired or does not have the correct permission -> github.com/Azure/azure-storage-azcopy/vendor/github.com/Azure/azure-storage-blob-go/azblob.NewResponseError, /go/src/github.com/Azure/azure-storage-azcopy/vendor/github.com/Azure/azure-storage-blob-go/azblob/zz_generated_response_error.go:29
===== RESPONSE ERROR (ServiceCode=AuthorizationPermissionMismatch) =====
Description=This request is not authorized to perform this operation using this permission.
RequestId:c4209a39-b01e-0060-01f1-c2ecc1000000
Time:2019-02-12T16:37:08.1763825Z, Details: 
   Code: AuthorizationPermissionMismatch
   PUT [...]
   Authorization: REDACTED
   Content-Length: [8388608]
   User-Agent: [AzCopy/10.0.7-Preview Azure-Storage/0.4 (go1.10.3; linux)]
   X-Ms-Client-Request-Id: [fd1a2db8-0c74-458e-5cac-e787f56a3aad]
   X-Ms-Version: [2018-03-28]
   --------------------------------------------------------------------------------
   RESPONSE Status: 403 This request is not authorized to perform this operation using this permission.
   Content-Length: [279]
   Content-Type: [application/xml]
   Date: [Tue, 12 Feb 2019 16:37:07 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Error-Code: [AuthorizationPermissionMismatch]
   X-Ms-Request-Id: [c4209a39-b01e-0060-01f1-c2ecc1000000]
   X-Ms-Version: [2018-03-28]
```

Then I ran it again and got:
```
Using OAuth token for authentication.

Job cde3f1df-4f14-f84d-609c-77102577c08d has started

Log file is located at: /home/jobevers/.azcopy/cde3f1df-4f14-f84d-609c-77102577c08d.log
Sync has enumerated 36 files to delete from destination. Do you want to delete these files ? Please confirm with y/n: y
583 Done, 0 Failed, 95 Pending, 678 Total, 2-sec Throughput (Mb/s): 6701.6566 

Job cde3f1df-4f14-f84d-609c-77102577c08d summary
Elapsed Time (Minutes): 0.2669
Total Number Of Copy Transfers: 642
Total Number Of Delete Transfers: 36
Number of Copy Transfers Completed: 642
Number of Copy Transfers Failed: 0
Number of Delete Transfers Completed: 36
Number of Delete Transfers Failed: 0
Final Job Status: Completed
```
I thought it was weird that 36 files had managed to transfer the first time, so I tried a third time to see what would happen:
```
Using OAuth token for authentication.

Job 2be6dc02-1d1a-7b47-47e9-98b4df5d8770 has started

Log file is located at: /home/jobevers/.azcopy/2be6dc02-1d1a-7b47-47e9-98b4df5d8770.log
Sync has enumerated 627 files to delete from destination. Do you want to delete these files ? Please confirm with y/n: n
642 File Scanned at Source, 627 Files Scanned at Destination
Authentication Failed. The SAS is not correct or expired or does not have the correct permission -> github.com/Azure/azure-storage-azcopy/vendor/github.com/Azure/azure-storage-blob-go/azblob.NewResponseError, /go/src/github.com/Azure/azure-storage-azcopy/vendor/github.com/Azure/azure-storage-blob-go/azblob/zz_generated_response_error.go:29
===== RESPONSE ERROR (ServiceCode=AuthorizationPermissionMismatch) =====
Description=This request is not authorized to perform this operation using this permission.
RequestId:2ffea49d-701e-000d-57f2-c2588a000000
Time:2019-02-12T16:48:33.2693489Z, Details: 
   Code: AuthorizationPermissionMismatch
   PUT [...]
   Authorization: REDACTED
   Content-Length: [2815385]
   User-Agent: [AzCopy/10.0.7-Preview Azure-Storage/0.4 (go1.10.3; linux)]
   X-Ms-Client-Request-Id: [2fc77bed-72f7-41d7-7407-9105768a1dcb]
   X-Ms-Version: [2018-03-28]
   --------------------------------------------------------------------------------
   RESPONSE Status: 403 This request is not authorized to perform this operation using this permission.
   Content-Length: [279]
   Content-Type: [application/xml]
   Date: [Tue, 12 Feb 2019 16:48:32 GMT]
   Server: [Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0]
   X-Ms-Error-Code: [AuthorizationPermissionMismatch]
   X-Ms-Request-Id: [2ffea49d-701e-000d-57f2-c2588a000000]
   X-Ms-Version: [2018-03-28]
```

2) Why am I being prompted to delete files on the destination.  Isn't the point of sync to only copy newer files?

### How can we reproduce the problem in the simplest way?

Run sync?

### Have you found a mitigation/solution?

No.

**Comment by zezha-msft** [comment]
Hi @jobevers, thanks for reaching out!

Sync replicates the source's content to the destination, this means that the extra files at the destination may need to be deleted in order to match the source. Please refer to this wiki page on a brief introduction.

Did you target the same container in all three invocations? Was there any permission change to your account?

Btw the sync command was re-written and will be shipped with the next release.

**Comment by zezha-msft** [comment]
Hi @jobevers, the sync command was rewritten and released in v10.0.8. Please give the new version a try and let us know if you still encounter any problem.

Here is a [wiki](https://github.com/Azure/azure-storage-azcopy/wiki/Sync-command-guide) explaining how the new sync command works. Please feel free to reach out if you have any question or concern.
